src/platform/packages/private/kbn-grid-layout/grid/use_grid_layout_events/panel/state_manager_actions.ts

```diff
@@ -8,134 +8,148 @@
+  type: Omit<ActivePanelEvent['type'], 'click'>,
+  gridLayoutStateManager: GridLayoutContextType['gridLayoutStateManager'],
   pointerPixel: PointerPosition,
   lastRequestedPanelPosition: MutableRefObject<GridPanelData | undefined>
 ) => {
   const {
     runtimeSettings$: { value: runtimeSettings },
-    interactionEvent$,
-    proposedGridLayout$,
-    activePanel$,
-    rowRefs: { current: gridRowElements },
+    activePanelEvent$,
+    gridLayout$,
+    layoutRef: { current: gridLayoutElement },
+    headerRefs: { current: gridHeaderElements },
+    sectionRefs: { current: gridSectionElements },
   } = gridLayoutStateManager;
-  const interactionEvent = interactionEvent$.value;
-  if (!interactionEvent || !runtimeSettings || !gridRowElements) {
+  const activePanel = activePanelEvent$.value;
+  const currentLayout = gridLayout$.value;
+  if (!activePanel || !runtimeSettings || !gridSectionElements || !currentLayout) {
     // if no interaction event return early
     return;
   }
-
-  const currentLayout = proposedGridLayout$.value;
-
-  const currentPanelData = currentLayout?.[interactionEvent.targetRow].panels[interactionEvent.id];
-
+  const currentPanelData = currentLayout[activePanel.targetSection].panels[activePanel.id];
   if (!currentPanelData) {
     return;
   }
 
-  const isResize = interactionEvent.type === 'resize';
+  const { columnCount, gutterSize, rowHeight, columnPixelWidth } = runtimeSettings;
+  const isResize = activePanel.type === 'resize';
 
   const previewRect = (() => {
     if (isResize) {
       const layoutRef = gridLayoutStateManager.layoutRef.current;
-      const maxRight = layoutRef
-        ? layoutRef.getBoundingClientRect().right - runtimeSettings.gutterSize
-        : window.innerWidth;
-      return getResizePreviewRect({ interactionEvent, pointerPixel, maxRight });
+      const maxRight = layoutRef ? layoutRef.getBoundingClientRect().right : window.innerWidth;
+      return getResizePreviewRect({ activePanel, pointerPixel, maxRight });
     } else {
-      return getDragPreviewRect({ interactionEvent, pointerPixel });
+      return getDragPreviewRect({ activePanel, pointerPixel });
     }
   })();
 
-  activePanel$.next({ id: interactionEvent.id, position: previewRect });
-
-  const { columnCount, gutterSize, rowHeight, columnPixelWidth } = runtimeSettings;
-
   // find the grid that the preview rect is over
-  const lastRowId = interactionEvent.targetRow;
-  const targetRowId = (() => {
-    // TODO: temporary blocking of moving with keyboard between sections till we have a better way to handle keyboard events between rows
-    if (isResize || isKeyboardEvent(e)) return lastRowId;
-    const previewBottom = previewRect.top + rowHeight;
+  const lastSectionId = activePanel.targetSection;
+  let previousSection;
+  let targetSectionId: string | undefined = (() => {
+    // TODO: temporary blocking of moving with keyboard between sections till we can fix the bug with commiting the action
```

Comment:
I tried un-commenting this, but for some reason, keyboard actions get blocked once a panel changes sections. It's probably a minor issue but, since this PR is already complicated enough, I think it's best if we leave this as a follow-up